pkg/logs/pipeline/provider.go

```diff
@@ -166,12 +166,16 @@ func (p *provider) NextPipelineChan() chan *message.Message {
 
 // Flush flushes synchronously all the contained pipeline of this provider.
 func (p *provider) Flush(ctx context.Context) {
-	for _, p := range p.pipelines {
+
+	for _, pipeline := range p.pipelines {
 		select {
 		case <-ctx.Done():
 			return
 		default:
-			p.Flush(ctx)
+			pipeline.Flush(ctx)
 		}
 	}
+	if p.serverless {
+		<-p.outputChan
+	}
```

Comment:
Differing from my prev commits where I assumed only one pipeline was in use in Serverless, I'm now using of the no-op auditor output channel to sync a flush with a payload being sent to intake